No fine (GPS) position on Android anymore

For some time now I don't get the PositionService working anymore, so that it would provide me with periodic position updates from GPS. When I start the service I get one position and that's it. Also the GPS symbol, which normally appears in the status bar for any app which activates GPS, does not show up anymore. I have provided a slightly modified HelloGluon version which should show the issue. This behaviour is consistent on all my devices and I even made sure that they had a valid GPS fix before I started the test. I even provided my own AndroidManifest.xml file (is that actually still necessary?) requesting coarse and fine location, but it did not change anything. So what is going on here? I tested this with client-maven-plugin 0.1.33 as well as 0.1.34-SNAPSHOT and Attach 4.0.9 as well as 4.0.10-SNAPSHOT on my Samsung tablet (Android 10) and phone (Android 8).
[HelloGluon.zip](https://github.com/gluonhq/attach/files/5453502/HelloGluon.zip)

I investigated this issue a little bit further. I modified the DalvikPositionService
so that it explicitly requests a "gps" location provider if gpsEnabled in the constructor
is true. The result is that I get the following exception when I later stop the service
again. (The delay of the exception is probably caused by the fact that the looperTask
does not do any exception handling.)

```
11-21 16:08:00.715 30811 32285 D GluonAttach: Stop listening for location changes
11-21 16:08:00.715 30811 32285 D GraalGluon: ATTACH_DALVIK, tid = 32285, existed? 1, dalvikEnv at 0x7d79d7b100
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] JNI DETECTED ERROR IN APPLICATION: JNI CallVoidMethod called with pending exception java.lang.SecurityException: "gps" location provider requires ACCESS_FINE_LOCATION permission.
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at java.lang.Exception android.os.Parcel.createException(int, java.lang.String) (Parcel.java:2088)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at void android.os.Parcel.readException(int, java.lang.String) (Parcel.java:2056)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at void android.os.Parcel.readException() (Parcel.java:2004)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at android.location.Location android.location.ILocationManager$Stub$Proxy.getLastLocation(android.location.LocationRequest, java.lang.String) (ILocationManager.java:1374)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at android.location.Location android.location.LocationManager.getLastKnownLocation(java.lang.String) (LocationManager.java:1433)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at void com.gluonhq.helloandroid.DalvikPositionService.initialize() (DalvikPositionService.java:186)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]   at void com.gluonhq.helloandroid.DalvikPositionService.start(long, float, boolean) (DalvikPositionService.java:104)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] Caused by: android.os.RemoteException: Remote stack trace:
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 	at com.android.server.LocationManagerService.checkResolutionLevelIsSufficientForProviderUseLocked(LocationManagerService.java:2299)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 	at com.android.server.LocationManagerService.getLastLocation(LocationManagerService.java:3080)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 	at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:594)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 	at android.os.Binder.execTransactInternal(Binder.java:1056)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 	at android.os.Binder.execTransact(Binder.java:1029)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] (Throwable with no stack trace)
11-21 16:08:00.715 30811 32285 F ples.hellogluo: java_vm_ext.cc:570] 
11-21 16:08:00.716 30811 32285 F ples.hellogluo: java_vm_ext.cc:570]     in call to CallVoidMethod
```

This is strange because in the actually used manifest the requested permission is granted.
I verified this with the apkanalyzer like this:

```
$ JAVA_HOME=$JAVA8_HOME ~/Library/Android/sdk/tools/bin/apkanalyzer manifest print HelloGluon-3.apk 
<?xml version="1.0" encoding="utf-8"?>
<manifest
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:versionCode="1"
    android:versionName="1.0"
    android:compileSdkVersion="29"
    android:compileSdkVersionCodename="10"
    package="com.gluonhq.samples.hellogluon"
    platformBuildVersionCode="29"
    platformBuildVersionName="10">

    <uses-sdk
        android:minSdkVersion="21"
        android:targetSdkVersion="29" />

    <supports-screens
        android:xlargeScreens="true" />

    <uses-permission
        android:name="android.permission.INTERNET" />

    <uses-permission
        android:name="android.permission.ACCESS_FINE_LOCATION" />

    <uses-permission
        android:name="android.permission.READ_EXTERNAL_STORAGE" />

    <uses-permission
        android:name="android.permission.WRITE_EXTERNAL_STORAGE" />

    <application
        android:label="HelloGluon"
        android:icon="@ref/0x7f080000"
        android:debuggable="true"
        android:appComponentFactory="android.support.v4.app.CoreComponentFactory">

        <activity
            android:name="com.gluonhq.helloandroid.MainActivity"
            android:configChanges="0xa0">

            <intent-filter>

                <category
                    android:name="android.intent.category.LAUNCHER" />

                <action
                    android:name="android.intent.action.MAIN" />
            </intent-filter>
        </activity>

        <activity
            android:name="com.gluonhq.helloandroid.PermissionRequestActivity" />
    </application>
</manifest>
```

Also in the dialogue, which pops up when I launch the app for the first time, I grant
the permission.

So there seems to be something wrong with the permission handling but I have no idea what.

--- a/modules/position/src/main/native/android/dalvik/DalvikPositionService.java
+++ b/modules/position/src/main/native/android/dalvik/DalvikPositionService.java
@@ -73,14 +73,17 @@ public class DalvikPositionService implements LocationListener {
     private final boolean debug;
 
     public DalvikPositionService(Activity activity) {
-
         this.activityContext = activity;
         debug = Util.isDebug();
         if (debug) {
             Log.v(TAG, "Construct DalvikPositionService");
         }
-        boolean gpsEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION) | // Both permissions require to be verified
-                Util.verifyPermissions(Manifest.permission.ACCESS_FINE_LOCATION);
+
+        // Verify both permissions
+        boolean coarseEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION);
+        boolean fineEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_FINE_LOCATION);
+        boolean gpsEnabled = coarseEnabled || fineEnabled;
+
         if (!gpsEnabled) {
             Log.v(TAG, "GPS disabled. ACCESS_COARSE_LOCATION or ACCESS_FINE_LOCATION permissions are required");
         }

--- a/modules/position/src/main/native/android/dalvik/DalvikPositionService.java
+++ b/modules/position/src/main/native/android/dalvik/DalvikPositionService.java
@@ -79,7 +79,7 @@ public class DalvikPositionService implements LocationListener {
         if (debug) {
             Log.v(TAG, "Construct DalvikPositionService");
         }
-        boolean gpsEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION) | // Don't use || here! 
+        boolean gpsEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION) | // Both permissions require to be verified
                 Util.verifyPermissions(Manifest.permission.ACCESS_FINE_LOCATION);
         if (!gpsEnabled) {
             Log.v(TAG, "GPS disabled. ACCESS_COARSE_LOCATION or ACCESS_FINE_LOCATION permissions are required");

--- a/modules/position/src/main/native/android/dalvik/DalvikPositionService.java
+++ b/modules/position/src/main/native/android/dalvik/DalvikPositionService.java
@@ -79,7 +79,7 @@ public class DalvikPositionService implements LocationListener {
         if (debug) {
             Log.v(TAG, "Construct DalvikPositionService");
         }
-        boolean gpsEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION) || 
+        boolean gpsEnabled = Util.verifyPermissions(Manifest.permission.ACCESS_COARSE_LOCATION) | // Don't use || here! 
                 Util.verifyPermissions(Manifest.permission.ACCESS_FINE_LOCATION);
         if (!gpsEnabled) {
             Log.v(TAG, "GPS disabled. ACCESS_COARSE_LOCATION or ACCESS_FINE_LOCATION permissions are required");

